src/nvim/spell.c

```diff
@@ -1344,7 +1344,8 @@ size_t spell_move_to(win_T *wp, int dir, bool allwords, bool curline, hlf_T *att
                             : p - buf) > wp->w_cursor.col)) {
             col = (colnr_T)(p - buf);
 
-            bool can_spell = decor_spell_nav_col(wp, lnum, &decor_lnum, col, &decor_error);
+            TriState res = decor_spell_nav_col(wp, lnum, &decor_lnum, col, &decor_error);
+            bool can_spell = TRISTATE_TO_BOOL(res, false);
```

Comment:
Should `spell = false` also prevent syntax spell checking and/or spell checking without `noplainbuffer`.

Reply:
I think it's something like:

| `provider` | `has_syntax` | `noplainbuffer` | `can_spell` |
|-|-|-|-|
| `1`             | `x`           | `x`                     | `1`               |
| `0`             | `1`           | `x`                     | `1`               |
| `0`             | `0`           | `0`                     | `1`               |
| `0`             | `0`           | `1`                     | `0`               |

Reply:
Yes